Generation/src/components/PythiaInterface.cpp

```diff
@@ -54,6 +66,51 @@ StatusCode PythiaInterface::initialize() {
   m_iAbort = 0;
   m_iEvent = 0;
 
+  // Begin ME/PS Matching specific code
+  // Check if jet matching should be applied.
+  m_doMePsMatching = m_pythiaSignal->settings.flag("JetMatching:merge");
+
+  // Check if internal merging should be applied.
+  m_doMePsMerging = !(m_pythiaSignal->settings.word("Merging:Process").compare("void") == 0);
+
+  // Currently, only one scheme at a time is allowed.
+  if (m_doMePsMerging && m_doMePsMatching) {
+    return Error ( "Jet matching and merging cannot be used simultaneously!" );
+  }
+  
+  // Allow to set the number of additional partons dynamically.
+  if (m_doMePsMerging) {
+    // Store merging scheme.
+    int scheme = ( m_pythiaSignal->settings.flag("Merging:doUMEPSTree")
+                || m_pythiaSignal->settings.flag("Merging:doUMEPSSubt")) ?
+                1 :
+                 ( ( m_pythiaSignal->settings.flag("Merging:doUNLOPSTree")
+                || m_pythiaSignal->settings.flag("Merging:doUNLOPSSubt")
+                || m_pythiaSignal->settings.flag("Merging:doUNLOPSLoop")
+                || m_pythiaSignal->settings.flag("Merging:doUNLOPSSubtNLO")) ?
+                2 :
+                0 );
+    m_setting = std::unique_ptr<Pythia8::amcnlo_unitarised_interface>(new Pythia8::amcnlo_unitarised_interface(scheme));
+    m_pythiaSignal->setUserHooksPtr(m_setting.get());
+  }
+
+  // For jet matching, initialise the respective user hooks code.
+  if (m_doMePsMatching) {
+    m_matching = std::unique_ptr<Pythia8::JetMatchingMadgraph>(new Pythia8::JetMatchingMadgraph());
+    if (!m_matching) {
+      return Error ( " Failed to initialise jet matching structures.");
+    }
+    m_pythiaSignal->setUserHooksPtr(m_matching.get());
+  }
+
+  // store additional branch containing matching/merging validation observables
+  if(m_doMePsMatching || m_doMePsMerging) declareOutput("mePsMatchingVars" , m_handleMePsMatchingVars, "mePsMatchingVars");
```

Comment:
I'd prefer braces and a new line after the if statement - reducing the line length.